packages/utils/src/promise.ts

```diff
@@ -80,14 +80,12 @@ export async function racePromisesWithCutoff<T>(
 ): Promise<(Error | T)[]> {
   // start the cutoff and timeout timers
   let cutoffObserved = false;
-  const cutoffPromise = new Promise((_resolve, reject) => setTimeout(reject, cutoffMs)).catch((e) => {
+  const cutoffPromise = new Promise((resolve) => setTimeout(resolve, cutoffMs)).then(() => {
```

Comment:
@g11tech  wanted keep the diff as minimal as possible but we could also just do now, maybe makes the code more readble

```
 const cutoffPromise = sleep(cutoffMs).then(() => {
    cutoffObserved = true;
  });
```

Reply:
Yeah, I think thats more readable